packages/appsync-modelgen-plugin/src/utils/process-connections-v2.ts

```diff
@@ -129,14 +129,15 @@ export function processConnectionsV2(
   model: CodeGenModel,
   modelMap: CodeGenModelMap,
   shouldUseModelNameFieldInHasManyAndBelongsTo: boolean = false,
-  isCustomPKEnabled: boolean = false
+  isCustomPKEnabled: boolean = false,
+  shouldUseFieldsInAssociatedWithInHasOne: boolean = false,
```

Comment:
Naming n.p. but you probably want `Within` instead of `WithIn` here.

Reply:
This should be read as "should use fields in associatedWith in hasOne".